### PR TITLE
Implement the game phase and turn controller (Needs to integrate to GUI)

### LINKED ISSUE #95: Implement the Game Phase and Turn Controller
### Description

The repository has a `GamePhase` field, but no service owns phase transitions
or validates when actions are legal. Add one orchestration service responsible
for the game state machine and current-player advancement.

Gameplay services should perform their own rule calculations, while this
controller coordinates them.

### Recommended Classes and Functions

Create `service.GameFlowService`:

```java
public void requireActiveGame(GameState gameState)
public void requirePhase(GameState gameState, GamePhase requiredPhase)
public void beginTurn(GameState gameState)
public void completeReinforcementPhase(GameState gameState)
public void completeAttackPhase(GameState gameState)
public void completeFortifyPhase(GameState gameState)
public void endTurn(GameState gameState)
public Player findNextActivePlayer(GameState gameState)
```

Add `ENDED` to `GamePhase`, or consistently use `gameFinished` to reject all
further actions. Adding `ENDED` is recommended because it makes the state
machine explicit.

### Acceptance Criteria

- Legal progression is `REINFORCEMENT -> ATTACK -> FORTIFY`.
- Completing fortification ends the turn.
- The next player comes from `turnOrder` and must also be active.
- Turn order wraps from the last active player to the first.
- Eliminated players are skipped.
- Actions are rejected when the game is finished.
- Invalid phase transitions are rejected.
- Beginning a new turn starts reinforcement processing.

### Required Tests

Create `GameFlowServiceTest`:

- `completeReinforcementPhase_movesToAttack`
- `completeAttackPhase_movesToFortify`
- `completeFortifyPhase_endsTurnAndMovesToReinforcement`
- `endTurn_advancesToNextActivePlayer`
- `endTurn_wrapsToFirstActivePlayer`
- `endTurn_skipsEliminatedPlayer`
- `requirePhase_rejectsWrongPhase`
- `requireActiveGame_rejectsFinishedGame`
- `invalidPhaseTransition_isRejected`

### PR DESCRIPTION
**Pull request by YC-Qiu**
This pull request introduces a new `GameFlowService` class to coordinate legal phase transitions and current-player advancement in the game, along with a comprehensive test suite to ensure correct behavior. The changes establish clear rules for phase progression, player turns, and error handling for invalid transitions.

**Core game flow orchestration:**

* Added the `GameFlowService` class, which manages phase transitions (reinforcement, attack, fortify), validates game and phase states, and advances the current player according to game rules. This includes enforcement of legal transitions and handling of eliminated players.

**Testing and validation:**

* Introduced `GameFlowServiceTest`, a suite of unit tests covering all major game phase transitions, player advancement, error cases for invalid transitions, and edge cases such as skipping eliminated players and handling endgame states.

!!!! To integrate to UI, need Attack phase, reinforcement phase, fortify phase done first.

### CODE REVIEW

File: src/main/java/service/GameFlowService.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                    "Turn order and active players must not be empty"
+            );
+        }
+        if (currentPlayerIndex < 0) {
+            throw new IllegalStateException(
+                    "Current player must be in the turn order"
+            );
+        }
+
+        for (int offset = 1; offset <= turnOrder.size(); offset++) {
```
Comment (doc4science) [inline src/main/java/service/GameFlowService.java]: offset not particularly descriptive

**Review by doc4science — APPROVED** [review]
looks good, minor feedback, but good overall.

File: src/main/java/service/GameFlowService.java (hunk 1 above)
Comment (YC-Qiu) [reply src/main/java/service/GameFlowService.java]: Changed to `positionsAhead`
